pkg/codegen/dotnet/gen_program.go

```diff
@@ -34,6 +34,8 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
+var ProjectKeysToOmit = []string{"configuration", "resources", "outputs"}
```

Comment:
I think we should be omitting everything except the keys pulumi explicitly knows about.

Reply:
I think we should respect `project.AdditionalKeys`. I would put this change in `"github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/codegen".Eject`.

Reply:
hmm yes that's probably a better call actually, I didn't realize eject had a chance to change the project.